rudder-web/src/main/scala/com/normation/rudder/web/rest/RestExtractorService.scala

```diff
@@ -679,5 +713,27 @@ case class RestExtractorService (
     }
   }
 
+  def extractQuery (params : Map[String,List[String]]) : Box[Option[Query]] = {
+    extractOneValue(params,"query")(convertToQuery) match {
+      case Full(None) =>
+        extractOneValue(params,CRITERIA)(convertToQueryCriterion) match {
+          case Full(None) => Full(None)
+          case Full(Some(criterion)) =>
+            for {
+              optType <- extractOneValue(params,TARGET)(convertToQueryReturnType)
```

Comment:
I'm pretty sure that it is not what we want, because with that, in case of a bad extract for target, we get a Full(none) that buble up to ignoring the query and returning the full list of nodes. 
This is extremelly hard to understand that Rudder ignored your request. 
In fact, each time the user gives (valid) param names, we should assure that we are not ignoring them - if it bored itself to specify a rather complexe query object, he DOES want to know that the string he gaves what not OK. 

This place is only the tip of the iceberg, there is other places that behaves the same. Our motto should be "fail early, fail explicitly".